### PR TITLE
DM-40418: Release Management GitHub Action

### PR DESCRIPTION
**Pull request by dspeck1**
Add GitHub Action to build releases.   Playbook is updated to include release management instructions.  This also moves the base Dockerfile into the base directory and changes naming of activator Dockerfile.  Both the build base and build GitHub actions are updated accordingly.

### CODE REVIEW

File: doc/playbook.rst
Comment (kfindeisen) [inline doc/playbook.rst:79]: We might want to require some standard information in the release description (possibly more a question for @ktlim). For my part, I'd want to record the underlying Science Pipelines version (especially if it can't be stored in the container as a label!), because that's something that _everybody_ will be asking about Prompt Processing.

**Review by kfindeisen — CHANGES_REQUESTED** [review]
Sorry for the delay, it took me a while to digest things (and seeing the workflow in action on lsst-dm/next_visit_fan_out#9 was indeed useful).

My main concern is the unclear relationship between `ci-release` and `build-service`, and the policy decisions that follow from that. Is `ci-release` meant to replace `build-service` (if so, why did you create a new script from scratch instead of just injecting support for tag refs into the old one?), or is it meant to be a supplement (if so, why does it run on anything other than tags/releases)? I'd prefer the former, just to be sure that we're really building things the same way for both testing and production.

Either way, we need to clean up how `latest` base containers are handled before a release workflow that always uses `latest` is safe to merge.

File: .github/workflows/ci-release.yaml
Comment (dspeck1) [reply .github/workflows/ci-release.yaml:16]: This was to get an initial workflow in place so we could do testing.  To show and run in GitHub the workflow needs to be pushed to main.  There will be changes.  This can replace `build-service`, but didn't want to do that before testing was completed.

File: Dockerfile
Comment (dspeck1) [reply Dockerfile:1]: The name of the Dockerfile is not an input parameter to the composite GitHub Action.  It is a best practice to give each container its own sub-directory and to not re-name the Dockerfile.

File: Dockerfile
Comment (dspeck1) [reply Dockerfile:6]: Ok, thanks.  I will remove.

File: Dockerfile
Comment (kfindeisen) [reply Dockerfile:1]: Good to know, but you didn't create subdirectories either.

I don't think it's a good idea to have two identical Dockerfiles.

Comment (dspeck1) [reply Dockerfile:1]: I agree.  The two identical Dockerfiles are there for co-existence.  

I will update the base container workflow and add a sub directory.

File: .github/workflows/ci-release.yaml
Comment (kfindeisen) [reply .github/workflows/ci-release.yaml:16]: Sorry, reading this again I don't follow. You seem to have already tested this before putting it in review (https://github.com/lsst-dm/prompt_processing/pkgs/container/prompt_processing shows the PR build, after all), and without pushing to `main`. Isn't the only part that must be tested on `main` the actual tag use?

Given your later remark about "co-existence", and the absence of a Phalanx PR, I feel like I'm missing something fundamental about your goal. Would it be useful to have a quick meeting on Zoom or Slack huddle? I have a meeting from 2-3 PST (4-5 CST) but am otherwise free.

Comment (dspeck1) [reply .github/workflows/ci-release.yaml:16]: Thanks.  Later in the day for me now.  Would a time work tomorrow?  I am free before noon PST or we could stay on the standup?

Comment (kfindeisen) [reply .github/workflows/ci-release.yaml:16]: After standup works for me.

File: doc/playbook.rst
Comment (kfindeisen) [reply doc/playbook.rst:79]: We discussed this at PP standup and came up with the following list:
- Any specific motivations for the release (for example, including a specific feature, fix, or compatibility change)
- Science Pipelines version and rubin-env version
- Changes to the APDB and Alerts schemas, to be replaced by version numbers once we have them.

I can go ahead and add the version numbers to the 1.0.0 description, since I can probably get those more easily than @dspeck1.

File: doc/playbook.rst
Comment (kfindeisen) [reply doc/playbook.rst:73]: For the purpose of distinguishing between major and minor versions, does compatibility mean:

For Prompt Processing,
* ability to read fanned-out `nextVisit` messages
* schema compatibility of APDB and Alert Stream outputs
* no particular requirements on Butler outputs

For fan-out,
* ability to read incoming Summit messages
* schema compatibility of fanned-out messages

The question of compatibility for fanned-out messages is a bit problematic because, so long as we're just dumping them into a Python dataclass, almost any change is incompatible: removing a field will give an error unless we include a default, as usual for schemas, but _adding_ a field will give an error unconditionally.

File: .github/workflows/ci-release.yaml
Comment (kfindeisen) [reply .github/workflows/ci-release.yaml:16]: We discussed this this afternoon -- the plan was to push the new workflow and Dockerfile alongside the old ones, then leave them there for an evaluation period while the old files remained untouched. They would then be consolidated on a later PR also associated with DM-40418. This goes against DM workflow, but it does explain the main issues I was confused about, so I won't hold up merge over it.

**Review by kfindeisen — APPROVED** [review]

File: Dockerfile
Comment (dspeck1) [reply Dockerfile:1]: Moved the base Dockerfile to a new base directory.  Removed the redundant Dockerfile.activator and updated  the build service GitHub Action.

File: doc/playbook.rst
Comment (dspeck1) [reply doc/playbook.rst]: Moved Release instructions to a subsection of Containers.  Thanks for suggestion!

File: doc/playbook.rst
Comment (dspeck1) [reply doc/playbook.rst]: Fixed link to pep.  Also updated instructions to use `X.Y.Z` for releases without prepending `v` to be consistent with other release numbering schemes used by LSST.

File: doc/playbook.rst
Comment (dspeck1) [reply doc/playbook.rst]: Added the notes above on release notes into the playbook.

File: doc/playbook.rst
Comment (kfindeisen) [reply doc/playbook.rst]: ??? My question was because it was a subsection of Containers to begin with...

File: .github/workflows/build-base.yml
Comment (kfindeisen) [inline .github/workflows/build-base.yml:47]: Please merge this change with the commit that created `base/`.

File: .github/workflows/build-base.yml
Comment (kfindeisen) [inline .github/workflows/build-base.yml:72]: Please merge with the commit that removed it; I thought the removal was deliberate!

Is `working-directory: base` still used in this mode?

**Review by kfindeisen — CHANGES_REQUESTED** [review]
Please restore the `PROMPT_PROCESSING_DIR` definition at the minimum; the service will not work if this is not an absolute path. I've made a few other suggestions as well (in particular, please [squash all fixup-type commits](https://developer.lsst.io/work/flow.html#rebase-commits-from-code-reviews-rather-than-having-review-feedback-commits) before final merge).

File: Dockerfile
Comment (dspeck1) [reply Dockerfile:5]: Ok earlier there was a request to remove those environment variables as they were Google specific.   Highlighting must not have been clear.  I'll add back in.

Comment (kfindeisen) [reply Dockerfile:5]: Sorry. Only the PUBSUB variable is Google-specific.

Comment (dspeck1) [reply Dockerfile]: Set back to how it was and only removed the `PUBSUB_VERIFICATION_TOKEN`

File: doc/playbook.rst
Comment (dspeck1) [reply doc/playbook.rst]: Moved release management to its own section.

File: doc/playbook.rst
Comment (dspeck1) [reply doc/playbook.rst]: Adjusted.

File: doc/playbook.rst
Comment (dspeck1) [reply doc/playbook.rst]: Adjusted.

File: .github/workflows/build-base.yml
Comment (dspeck1) [reply .github/workflows/build-base.yml:72]: Merged.  Yes working directory base is used.

File: doc/playbook.rst
Comment (kfindeisen) [inline doc/playbook.rst:81]:
```suggestion
* Any specific motivation for the release (for example, including a specific feature, preparing for a specific observing run))
```

**Review by kfindeisen — APPROVED** [review]

**Pull request merged**
